src/videoserver/lib/utils.py

```diff
@@ -161,44 +145,83 @@ def validate_document(document, schema, **kwargs):
 
 
 class VideoValidator(Validator):
+    def _normalize_coerce_crop_to_dict(self, value):
+        """Convert crop string (x,y,w,h) to dict
+        """
+        # cerberus run coercion func before validation rule
+        # if provided value is malformed, return original value (str) to validate against regex rule
+        try:
+            x, y, width, height = [int(item) for item in value.split(',')]
+            return {"x": x, "y": y, "width": width, "height": height}
+        except (TypeError, ValueError, AttributeError):
+            return value
+
+    def _is_malformed_format(self, field, value):
+        if type(value) != dict:
+            # avoid duplicate message
+            if len(self._errors) == 0:
+                # using cerberus string type will always fail after coerced
+                self._error(field, 'must be of string type')
+            return True
+        return False
+
     def _validate_allow_crop_width(self, limit, field, value):
         """Test allowed crop width range
         The rule's arguments are validated against this schema:
         {'min': 'limit[0]', 'max': 'limit[1]'}
         """
+        if self._is_malformed_format(field, value):
+            return
         if limit and len(limit) == 2:
             wmin, wmax = limit
-            if value['width'] < wmin:
-                self._error(field, "width is lesser than minimum allowed crop width")
-            if value['width'] > wmax:
-                self._error(field, "width is greater than maximum allowed crop width")
+            width = value.get('width', 0)
```

Comment:
Cerberus run this validation along with `regex` rule, so request like this is still passed to `_validate_allow_crop_width`
```json
{
	"crop": {"a": 1}
}
```
but it will fail because of default value (0)
```json
{
    "crop": [
        "width 0 is less than minimum allowed crop width (320)",
        "height 0 is less than minimum allowed crop height (180)"
    ]
}
```

Reply:
It fails when I PUT:
```
{
	"crop": {"width": 320, "height": 400}
}
```
or
```
{
	"crop": {"x": 320, "height": 400, "width": 320}
}
```
If I do 
```
{
	"crop": "1,1,640,480"
}
``` it works fine.

Reply:
yes, it is expected result